plugin.video.viaplay/resources/lib/kodihelper.py

```diff
@@ -0,0 +1,228 @@
+        if setting == 'true':
+            return True
+        elif setting == 'false':
+            return False
+        else:
+            return setting
+
+    def set_setting(self, key, value):
+        return self.get_addon().setSetting(key, value)
+
+    def log(self, string):
+        msg = '%s: %s' % (self.logging_prefix, string)
+        xbmc.log(msg=msg, level=xbmc.LOGDEBUG)
+
+    def get_country_code(self):
+        country_id = self.get_setting('site')
+        if country_id == '0':
+            country_code = 'se'
+        elif country_id == '1':
+            country_code = 'dk'
+        elif country_id == '2':
+            country_code = 'no'
+        else:
+            country_code = 'fi'
+
+        return country_code
+
+    def get_sub_lang(self):
+        sub_lang_id = self.get_setting('sub_lang')
+        if sub_lang_id == '0':
+            sub_lang = 'sv'
+        elif sub_lang_id == '1':
+            sub_lang = 'da'
+        elif sub_lang_id == '2':
+            sub_lang = 'no'
+        else:
+            sub_lang = 'fi'
+
+        return sub_lang
+
+    def dialog(self, dialog_type, heading, message=None, options=None, nolabel=None, yeslabel=None):
+        dialog = xbmcgui.Dialog()
+        if dialog_type == 'ok':
+            dialog.ok(heading, message)
+        elif dialog_type == 'yesno':
+            return dialog.yesno(heading, message, nolabel=nolabel, yeslabel=yeslabel)
+        elif dialog_type == 'select':
+            ret = dialog.select(heading, options)
+            if ret > -1:
+                return ret
+            else:
+                return None
+
+    def authorize(self):
+        try:
+            self.vp.validate_session()
+            return True
+        except self.vp.ViaplayError as error:
+            if not error.value == 'PersistentLoginError' or error.value == 'MissingSessionCookieError':
+                raise
+            else:
+                return self.device_registration()
+
+    def log_out(self):
+        confirm = self.dialog('yesno', self.language(30042), self.language(30043))
+        if confirm:
+            self.vp.log_out()
+            # send Kodi back to home screen
+            xbmc.executebuiltin('XBMC.Container.Update(path, replace)')
+            xbmc.executebuiltin('XBMC.ActivateWindow(Home)')
+
+    def device_registration(self):
+        """Presents a dialog with information on how to activate the device.
+        Attempts to authorize the device using the interval returned by the activation data."""
+        activation_data = self.vp.get_activation_data()
+        message = self.language(30039).format(activation_data['verificationUrl'], activation_data['userCode'])
+        dialog = xbmcgui.DialogProgress()
+        xbmc.sleep(200)  # small delay to prevent DialogProgress from hanging
+        dialog.create(self.language(30040), message)
+        secs = 0
+        expires = activation_data['expires']
+
+        while secs < expires:
```

Comment:
You can use `xbmc.Monitor().abortRequested()` in your control loop statement to ensure it always breaks when kodi wants to exit